################################################################################
SHORT DESCRIPTION: 
################################################################################
Use Linux 3.x kuid/kgid types instead of former general int.

################################################################################
LONG DESCRIPTION: 
################################################################################
See:
commit 7a4e7408c5cadb240e068a662251754a562355e3
Author: Eric W. Biederman <ebiederm@xmission.com>
Date:   Mon Nov 14 14:29:51 2011 -0800

    userns: Add kuid_t and kgid_t and associated infrastructure in uidgid.h

    Start distinguishing between internal kernel uids and gids and
    values that userspace can use.  This is done by introducing two
    new types: kuid_t and kgid_t.  These types and their associated
    functions are infrastructure are declared in the new header
    uidgid.h.
    
    Ultimately there will be a different implementation of the mapping
    functions for use with user namespaces.  But to keep it simple
    we introduce the mapping functions first to separate the meat
    from the mechanical code conversions.
    
    Export overflowuid and overflowgid so we can use from_kuid_munged
    and from_kgid_munged in modular code.
    
    Acked-by: Serge Hallyn <serge.hallyn@canonical.com>
    Signed-off-by: Eric W. Biederman <ebiederm@xmission.com>

################################################################################
REMOVAL 
################################################################################
This patch is required to build this ralink driver version against kernels
3.5.y and later.

################################################################################
UPSTREAM PLAN
################################################################################
None

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None

Index: 2011_0427_RT3572_Linux_STA_v2.5.0.0.DPO/include/os/rt_linux.h
===================================================================
--- 2011_0427_RT3572_Linux_STA_v2.5.0.0.DPO.orig/include/os/rt_linux.h	2011-04-27 04:22:21.000000000 +0200
+++ 2011_0427_RT3572_Linux_STA_v2.5.0.0.DPO/include/os/rt_linux.h	2015-06-22 11:04:36.801753158 +0200
@@ -286,8 +286,13 @@
 
 typedef struct _OS_FS_INFO_
 {
-	int				fsuid;
-	int				fsgid;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,5,0)
+	kuid_t		fsuid;
+	kgid_t		fsgid;
+#else
+	int		fsuid;
+	int		fsgid;
+#endif
 	mm_segment_t	fs;
 } OS_FS_INFO;
 
